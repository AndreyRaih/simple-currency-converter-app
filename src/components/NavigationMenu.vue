<template>
  <n-tabs
    :class="$style.Container"
    :value="selectedItem"
    @update:value="(value) => $emit('update:navigation', value)"
    type="line"
    size="large"
  >
    <n-tab
      v-for="(option, index) in menu"
      :key="index"
      :name="option.name"
    >
      {{ option.label }}
    </n-tab>
  </n-tabs>
</template>

<style lang="less" module>
.Container {
  margin-bottom: 36px;
}
</style>

<script lang="ts">
import { defineComponent, PropType } from 'vue'
import { NTabs, NTab } from 'naive-ui'
import { NavigationMenuOption } from '@/typings.d'

export default defineComponent({
  name: 'NavigationMenu',
  components: {
    NTabs,
    NTab
  },
  props: {
    selectedItem: {
      type: String,
      default: null
    },
    menu: {
      required: true,
      type: Array as PropType<NavigationMenuOption[]>,
      default: () => ([])
    }
  },
  emits: ['update:navigation']
})
</script>
