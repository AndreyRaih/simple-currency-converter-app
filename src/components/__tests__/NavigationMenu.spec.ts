import { shallowMount } from '@vue/test-utils'
import NavigationMenu from '../NavigationMenu.vue'
import { NTabs } from 'naive-ui'
/**
 * NOTE: `const defaultOptions: any` - any type has been added in order to fix typings issue in runtime
 * This workaround decribed here: https://github.com/vuejs/test-utils/issues/194#issuecomment-694167869
 */
const defaultOptions: any = {
  props: {
    selectedItem: 'test',
    menu: [{
      name: 'test',
      label: 'Test menu item'
    }]
  }
}

describe('NavigationMenu.vue', () => {
  describe('Rendering', () => {
    it('should renders correctly', () => {
      const wrapper = shallowMount(NavigationMenu, defaultOptions)
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Functional', () => {
    it('custom event @update:value should be handled correct', () => {
      const wrapper = shallowMount(NavigationMenu, defaultOptions)

      const updatedData = 'New test value'
      const navTabElement = wrapper.findComponent(NTabs)
      navTabElement.vm.$emit('update:value', updatedData)

      const updateNavigationEvent = wrapper.emitted('update:navigation')
      expect(updateNavigationEvent).toHaveLength(1)
      expect((updateNavigationEvent as unknown[])[0]).toEqual([updatedData])
    })
  })
})
