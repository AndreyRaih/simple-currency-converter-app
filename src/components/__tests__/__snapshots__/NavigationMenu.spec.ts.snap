// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu.vue Rendering should renders correctly 1`] = `
<n-tabs-stub
  addable="false"
  class="Container"
  closable="false"
  size="large"
  tabspadding="0"
  type="line"
  value="test"
/>
`;
